### PR TITLE
Coinify Refactor

### PR DESCRIPTION
**Pull request by plondon**
Depends on:
https://github.com/blockchain/bitcoin-exchange-client/pull/6
https://github.com/blockchain/bitcoin-coinify-client/pull/10

### CODE REVIEW

**Comment by plondon** [comment]
Depends on https://github.com/blockchain/bitcoin-coinify-client/pull/9

File: app/partials/coinify/summary.pug [hunk 1]
```diff
@@ -1,105 +1,80 @@
...
+      .flex-center(ng-show="(summaryForm.fiat.$error.max || summaryForm.fiat.$error.min) && !isBankTransfer()")
+        i.ti-alert.state-danger-text.mrs.type-h4
+        span.state-danger-text.mrs(ng-show="summaryForm.fiat.$error.max" translate="MAX_LIMIT_EXCEEDED" translate-values="{code: tempTrade.fiatCurrency, max: limits[vm.medium].max[tempTrade.fiatCurrency], medium: vm.medium}")
+        span.state-danger-text.mrs(ng-show="summaryForm.fiat.$error.min && !summaryForm.fiat.$error.max" translate="BELOW_MIN_LIMIT" translate-values="{code: tempTrade.fiatCurrency, min: limits[vm.medium].min[tempTrade.fiatCurrency], medium: vm.medium}")
+    .border.pam(ng-class="{'bg-grey': state.editAmount}")
+      .flex-row.flex-between.pam.border-bottom-light
+        span(ng-hide="isBankTransfer()" translate="BTC:")
+        span(ng-show="isBankTransfer()" translate="BTC_APPROX_RATE")
+        span.success(ng-hide="isBankTransfer()" uib-tooltip="{{'RECEIVE_BITCOIN_TO' | translate:{label: label} }}") {{ format(fromSatoshi(trade.BTCAmount || 0, bitcoin), bitcoin, true) }}
```
Comment (Sjors) [inline app/partials/coinify/summary.pug]: I would suggest creating a filter, so you can do e.g. `trade.BTCAount | myMagicFormat:...:...`. It's easier to write test cases for a filter than for view code.

File: app/templates/buy-quick-start.pug [hunk 2]
```diff
@@ -1,12 +1,12 @@
 .quick-start.col-md-6.col-sm-12.col-xs-12.pln.prn
   .pt-30.pb-30.border-desktop(ng-class="{'ph-15': !inMobileBuy}")
-    form(name="fiatForm" ng-submit="!status.busy && fiatForm.$valid && triggerBuy()" novalidate)
+    form(name="fiatForm" ng-submit="!status.busy && fiatForm.$valid && buy({quote: quote})" novalidate)
       .flex-row.flex-between
         .flex-center
           label.em-500(translate="ENTER_AMOUNT:")
-          helper-button.mbs(ng-disabled="false" ng-show="limits.card.min && limits.card.max" content="MAX_MIN_HELPER" values="{cardMax: limits.card.max, min: limits.card.min, bankMax: limits.bank.max, symbol: currencySymbol.symbol}")
+          helper-button.mbs(ng-disabled="false" ng-show="exchangeRate.fiat" content="MAX_MIN_HELPER" values="{max: limits.absoluteMax(transaction.currency.code), min: limits.absoluteMin(transaction.currency.code), code: transaction.currency.code}")
```
Comment (Sjors) [inline app/templates/buy-quick-start.pug]: Maybe move that complex `max:` `min:` object to the controller?
`content="MAX_MIN_HELPER" values = "$ctrl. ..."`?

**Comment by prwelber** [comment]
depends on: 
https://github.com/blockchain/bitcoin-coinify-client/pull/7
https://github.com/blockchain/bitcoin-exchange-client/pull/4

File: app/partials/coinify/summary.pug (hunk 1 above)
Comment (jtormey) [inline app/partials/coinify/summary.pug]: We have a filter that does exactly this: `{{ trade.BTCAmount | format: bitcoin: true }}`

**Comment by Sjors** [comment]
💯

**Comment by Sjors** [comment]
This also requires a new my-wallet-v3 release to work.

**Pull request merged**